lib/graphql_metrics/analyzer.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+# Execution order:
+# When used as instrumentation, an analyzer and tracing, the order of execution is:
+
+# * Tracer.capture_parsing_time
+# * Instrumentation.before_query (context setup)
+# * Tracer.capture_validation_time (twice, once for `analyze_query`, then `analyze_multiplex`)
+# * Analyzer#initialize (bit more context setup, instance vars setup)
+# * Analyzer#result
+# * Tracer.trace_field (n times)
+# * Instrumentation.after_query (call query and field callbacks, now that we have all static and runtime metrics
+#   gathered)
+# * Analyzer#extract_query
+# * Analyzer#query_extracted
+# * Analyzer#extract_fields_with_runtime_metrics
+#   * calls Analyzer#field_extracted n times
+#
+# When used as a simple analyzer, which doesn't gather or emit any runtime metrics (timings, arg values):
+# * Analyzer#initialize
+# * Analyzer#field_extracted n times
+# * Analyzer#result
+# * Analyzer#extract_query
+# * Analyzer#query_extracted
+
+module GraphQLMetrics
+  class Analyzer < GraphQL::Analysis::AST::Analyzer
+    def initialize(query_or_multiplex)
+      super
+
+      query_or_multiplex.context.namespace(CONTEXT_NAMESPACE).tap do |ns|
+        ns[ANALYZER_INSTANCE_KEY] = self
+      end
+
+      @query = query_or_multiplex
+      @static_query_metrics = nil
+      @static_field_metrics = []
+    end
+
+    def analyze?
+      query.valid? && query.context[GraphQLMetrics::SKIP_GRAPHQL_METRICS_ANALYSIS] != true
```

Comment:
Can't this be simplified to `query.valid? && !query.context[GraphQLMetrics::SKIP_GRAPHQL_METRICS_ANALYSIS]`? Rather than adding the `!= true`?

Reply:
Yeah that's better actually, since we may as well embrace that context value being truthy, rather than `true`.